Fix 'docker ps --help' so the options don't span more than one line

duglin opened the pull request:
and add a testcase to catch this in the future

Signed-off-by: Doug Davis dug@us.ibm.com

Commit:
Fix 'docker ps --help' so the options don't span more than one line
and add a testcase to catch this in the future.

While in there I also:
- removed extra periods from the few options that had them (new test)
- made the --filter option consistent across all command

Signed-off-by: Doug Davis <dug@us.ibm.com>

thaJeztah commented:
Just out of interest; multiple lines don't work with help-text? Or is it "not done" to have them?

duglin commented:
In a previous issue/PR we decided that the help text was getting messy and that all options should
be limited to just one line.

duglin commented:
If people want more info they should look at the more details help text, like the html or man pages.

thaJeztah commented on docs/sources/reference/commandline/cli.md:
```diff
@@ -1440,9 +1440,7 @@ The `docker rename` command allows the container to be renamed to a different na
-      -f, --filter=[]       Provide filter values. Valid filters:
-                              exited=<int> - containers with exit code of <int>
-                              status=(restarting|running|paused|exited)
+      -f, --filter=[]       Provide filter values.
```
Perhaps it would be clearer (and consistent) to write what it *does*, not what you *provide*, e.g. "Filter containers (by conditions)". Not a great text, perhaps you can come up with something better.

thaJeztah commented:
@duglin thanks for explaining. Added a suggestion

duglin commented:
@thaJeztah I updated the --filter per your suggestion.  My OCD really kicked in on that one  :-) and made the other --filter options for other commands consistent and noticed that some option text ended with a period, while others (most) did not - I fixed that too.  And added a test for it.

thaJeztah commented:
@duglin great! (I actually lol-ed at that, because I noted the periods, but didn't want to bring it up for this PR)

duglin commented:
@thaJeztah well, to complete the OCD-ness of it all, https://github.com/docker/docker/pull/11084 is going to work on ensuring we don't have extra spaces (especially at the end of the lines).

thaJeztah commented:
@duglin don't worry, I already saw that issue ✨

estesp commented:
LGTM

jfrazelle commented:
LGTM
